app/src/main/java/com/quran/labs/androidquran/widgets/HighlightingImageView.java

```diff
@@ -130,17 +152,178 @@ public void setNightMode(boolean isNightMode, int textBrightness) {
     adjustNightMode();
   }
 
-  public void highlightAyah(int sura, int ayah, HighlightType type) {
+  private void highlightFloatableAyah(Set<String> highlights, String currentSurahAyah) {
+    final Map<String, List<AyahBounds>> coordinatesData = ayahCoordinates.getAyahCoordinates();
+
+    String previousSurahAyah = currentSurahAyah;
+    for(String surahAyahs: highlights) {
+      previousSurahAyah = surahAyahs;
+    }
+    List<AyahBounds> startingBounds;
+    int arrowIndex = previousSurahAyah.indexOf("->");
+    if(arrowIndex != -1) {
+      // The ayah changed during animating
+      startingBounds = (List<AyahBounds>)animator.getAnimatedValue();
+      animator.cancel();
+      previousSurahAyah = previousSurahAyah.substring(arrowIndex + 2);
+    } else {
+      startingBounds = coordinatesData.get(previousSurahAyah);
+    }
+
+    final String ayahTransition = previousSurahAyah+"->"+currentSurahAyah;
+
+    // yes we make copies, because normalizing the bounds will change them
+    List<AyahBounds> previousAyahBoundsList = new ArrayList<>(startingBounds);
+    List<AyahBounds> currentAyahBoundsList = new ArrayList<>(coordinatesData.get(currentSurahAyah));
+
+    highlights.clear();
+    highlights.add(ayahTransition);
+
+    // TODO: reason whether this TypeEvaluator should be anonymous or not
```

Comment:
i'd vote for not because:
1. this code doesn't ideally belong in HighlightingImageView
2. you can write a nice unit test for it that way without needing a ui test just to test the behavior (i.e if your standalone class exposes these as functions, and the animation stuff is outside that class)

Reply:
Have taken the interpolator out of the HighlightingImageView class